caluma/caluma_core/management/commands/migrate_to_prefixed_apps.py

```diff
@@ -0,0 +1,109 @@
+                cursor.execute(query)
+            except ProgrammingError as e:  # pragma: no cover
+                # happens only on initial migration for new project
+                if e.args[0].startswith('relation "django_migrations" does not exist'):
+                    return False
+                raise
+            if cursor.fetchone():
+                return True
+        return False
+
+    def _get_table_list(self):
+        query = f"""SELECT tablename FROM pg_catalog.pg_tables WHERE tablename LIKE '{self._act_prefix}form_%' OR tablename LIKE '{self._act_prefix}workflow_%';"""
+        with connection.cursor() as cursor:
+            cursor.execute(query)
+            return [row[0] for row in cursor.fetchall()]
+
+    def _collect_content_type_queries(self):
+        queries = [
+            f"""UPDATE django_content_type SET app_label = CONCAT('{self.prefix}', app_label) WHERE app_label = 'form' OR app_label = 'workflow';"""
+        ]
+        if self.revert:
+            queries = [
+                f"""UPDATE django_content_type SET app_label = 'form' WHERE app_label = '{self.prefix}form';""",
+                f"""UPDATE django_content_type SET app_label = 'workflow' WHERE app_label = '{self.prefix}workflow';""",
+            ]
+        self.queries += queries
+
+    def _collect_django_migrations_queries(self):
+        queries = [
+            f"""UPDATE django_migrations SET app = CONCAT('{self.prefix}', app) WHERE app = 'form' OR app = 'workflow';"""
+        ]
+        if self.revert:
+            queries = [
+                f"""UPDATE django_migrations SET app = 'form' WHERE app = '{self.prefix}form';""",
+                f"""UPDATE django_migrations SET app = 'workflow' WHERE app = '{self.prefix}workflow';""",
+            ]
+        self.queries += queries
+
+    def _collect_prefix_tables_queries(self, tables: list):
+        for table in tables:
+            query = f"ALTER TABLE {table} RENAME TO {self.prefix}{table};"
+            if self.revert:
+                query = (
+                    f"ALTER TABLE {table} RENAME TO {table.replace(self.prefix, '')};"
+                )
+            self.queries.append(query)
+
+    def collect(self):
+        if not self._migration_needed():
+            return
+        self._collect_content_type_queries()
+        self._collect_django_migrations_queries()
+        tables = self._get_table_list()
+        self._collect_prefix_tables_queries(tables)
+
+    def apply(self):
+        with connection.cursor() as cursor:
+            for query in self.queries:
+                cursor.execute(query)
+
+    def handle(self, *args, **options):
+        if options["revert"]:
+            self.revert = True
+
+        self.collect()
+
+        if not self.queries:
+            self.stderr.write("Seems the DB is already in the desired state!")
+            return
+
+        self.stderr.write("\n".join(self.queries))
+
+        if options["force"]:
+            self.apply()
```

Comment:
I start with some highlevel nagging questions (playing dump):

Why is this magic needed? If the apps all have the right name and their migrations in their directory shouldn't django handle this?

Reply:
Exactly! Without those changes, django would create new tables, instead of renaming them.

The magic with the `force` argument exists, because in some deployments, we're not able to run django commands, but just sql scripts.

@anehx How are migrations run on those systems? With the overriden `migrate` command, everything will be taken care of.

Reply:
We'll need plain SQL scripts to run - no django commands.

Reply:
But how do you aplply migrations? Do you create sql scripts for them or run `manage.py migrate`?

Reply:
Oh, sorry didn't get what you mean.. The django migrations are runned with `manage.py migrate`

Reply:
Ok, so no need for an sql script in this case :tada:

@ganwell Is your question answered?

Reply:
Yes thanks. Go ahead....